src/engine/Input/Pointer.ts

```diff
@@ -260,34 +315,81 @@ export class Pointers extends Class {
    /**
     * Propogates events to actor if necessary
     */
-   public propogate(actor: any) {
-      var isUIActor = actor instanceof UIActor;
-      var i: number = 0,
-            len: number = this._pointerUp.length;
+   public propogate(actor: Actor) {
+      var isNotUIActor = !Actors.isUIActor(actor);
+      var i: number = 0;
+      var len: number = 0;
+      var pointerEvent: PointerEvent;
+      var pointer: Pointer;
+
+      i = 0;
+      len = this._pointerDown.length;
 
       for (i; i < len; i++) {
-         if (actor.contains(this._pointerUp[i].x, this._pointerUp[i].y, !isUIActor)) {
-            actor.eventDispatcher.emit('pointerup', this._pointerUp[i]);
+         pointerEvent = this._pointerDown[i];
+         pointer = pointerEvent.pointer;
+
+         if (actor.contains(pointer.lastWorldPos.x, pointer.lastWorldPos.y, isNotUIActor)) {
+            actor.eventDispatcher.emit('pointerdown', pointerEvent);
+
+            if (pointer.isDragStart && actor.capturePointer.captureDragEvents) {
+               actor.eventDispatcher.emit('pointerdragstart', pointerEvent);
+            }
          }
       }
 
       i = 0;
-      len = this._pointerDown.length;
+      len = this._pointerUp.length;
 
       for (i; i < len; i++) {
-         if (actor.contains(this._pointerDown[i].x, this._pointerDown[i].y, !isUIActor)) {
-            actor.eventDispatcher.emit('pointerdown', this._pointerDown[i]);
+         pointerEvent = this._pointerUp[i];
+         pointer = pointerEvent.pointer;
+
+         if (actor.contains(pointer.lastWorldPos.x, pointer.lastWorldPos.y, isNotUIActor)) {
+            actor.eventDispatcher.emit('pointerup', pointerEvent);
+
+            if (pointer.isDragEnd && actor.capturePointer.captureDragEvents) {
+               actor.eventDispatcher.emit('pointerdragend', pointerEvent);
+            }
          }
       }
 
       if (actor.capturePointer.captureMoveEvents) {
-
          i = 0;
          len = this._pointerMove.length;
 
          for (i; i < len; i++) {
-            if (actor.contains(this._pointerMove[i].x, this._pointerMove[i].y, !isUIActor)) {
-               actor.eventDispatcher.emit('pointermove', this._pointerMove[i]);
+            pointerEvent = this._pointerMove[i];
+            pointer = pointerEvent.pointer;
+
+            if (actor.contains(pointer.lastWorldPos.x, pointer.lastWorldPos.y, isNotUIActor)) {
```

Comment:
In general (not necessarily related to this PR), should we be worrying about Actor z-index too @eonarheim ?